test/e2e/helpers/kafka_helper.go

```diff
@@ -37,73 +42,81 @@ var (
 	topicGVR = schema.GroupVersionResource{Group: strimziApiGroup, Version: strimziApiVersion, Resource: strimziTopicResource}
 )
 
-func MustPublishKafkaMessage(client *testlib.Client, bootstrapServer string, topic string, key string, headers map[string]string, value string) {
+func MustPublishKafkaMessage(client *testlib.Client, selector map[string]string, topic string, key string, headers map[string]string, value string) {
 	cgName := topic + "-" + key
 
-	_, err := client.Kube.Kube.CoreV1().ConfigMaps(client.Namespace).Create(&corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      topic + "-" + key,
-			Namespace: client.Namespace,
-		},
-		Data: map[string]string{
-			"payload": value,
-		},
-	})
-	if err != nil {
-		client.T.Fatalf("Failed to create configmap %q: %v", cgName, err)
-		return
-	}
-
-	client.Tracker.Add(corev1.SchemeGroupVersion.Group, corev1.SchemeGroupVersion.Version, "configmap", client.Namespace, cgName)
-
-	var args []string
-	args = append(args, "-P", "-v", "-b", bootstrapServer, "-t", topic, "-k", key)
+	var kvlist []string
 	for k, v := range headers {
-		args = append(args, "-H", k+"="+v)
+		kvlist = append(kvlist, k+":"+v)
 	}
-	args = append(args, "/etc/mounted/payload")
 
-	client.T.Logf("Running kafkacat %v", args)
-
-	pod := corev1.Pod{
+	job := &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      cgName + "-producer",
 			Namespace: client.Namespace,
+			Labels:    selector,
 		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{{
-				Image:   "docker.io/edenhill/kafkacat:1.5.0",
-				Name:    cgName + "-producer-container",
-				Command: []string{"kafkacat"},
-				Args:    args,
-				VolumeMounts: []corev1.VolumeMount{{
-					Name:      "event-payload",
-					MountPath: "/etc/mounted",
-				}},
-			}},
-			RestartPolicy: corev1.RestartPolicyNever,
-			Volumes: []corev1.Volume{{
-				Name: "event-payload",
-				VolumeSource: corev1.VolumeSource{ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: cgName,
-					},
-				}},
-			}},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:  cgName + "-producer-container",
```

Comment:
I understand that you want to test your binding here, but maybe can we have it in another test? like another e2e test binding -> kafka source.
I would love to keep the kafkacat sender because it's a rock solid kafka client and helps to check the effective integration of kafka source with outside world.

Reply:
+1 on that @slinkydeveloper

Reply:
I'm happy to split this into two tests.  I may look at making this part of the table test, if that's plausible.  Will circle back after I do that.

Reply:
Alright, this should be done.  The table test was just awkward enough that it probably wasn't worth it.